test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIBase.java

```diff
@@ -164,4 +166,46 @@ protected void breakpointForCommunication() throws JDITestRuntimeException {
         throw new JDITestRuntimeException("** event '" + event + "' IS NOT a breakpoint **");
     }
 
+    // Waiting for ThreadStart and ThreadDeath event we can get events from system threads,
+    // so need to ensure we get the event from the desired thread.
+    protected void waitThreadStart(String threadName) throws JDITestRuntimeException {
```

Comment:
I think this API is misnamed. It's not just waiting for a ThreadStart for the specified thread. It's waiting for any event other than ThreadStart for some other thread or ThreadDeath for any thread. That's a bit of an odd combination, and certainly not obvious based on the method name and description.

Reply:
I reworked the fix. Hope now it's clearer.